contracts/libraries/VaultLifecycleSTETH.sol

```diff
@@ -0,0 +1,757 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.7.3;
+pragma experimental ABIEncoderV2;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {Vault} from "./Vault.sol";
+import {ISTETH, IWSTETH} from "../interfaces/ISTETH.sol";
+import {IWETH} from "../interfaces/IWETH.sol";
+import {ICRV} from "../interfaces/ICRV.sol";
+import {
+    IStrikeSelection,
+    IOptionsPremiumPricer
+} from "../interfaces/IRibbon.sol";
+import {GnosisAuction} from "./GnosisAuction.sol";
+import {
+    IOtokenFactory,
+    IOtoken,
+    IController,
+    GammaTypes
+} from "../interfaces/GammaInterface.sol";
+import {IERC20Detailed} from "../interfaces/IERC20Detailed.sol";
+import {SupportsNonCompliantERC20} from "./SupportsNonCompliantERC20.sol";
+
+library VaultLifecycleSTETH {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+    using SupportsNonCompliantERC20 for IERC20;
+
+    struct CloseParams {
+        address OTOKEN_FACTORY;
+        address USDC;
+        address currentOption;
+        uint256 delay;
+        uint256 lastStrikeOverride;
+        uint256 overriddenStrikePrice;
+    }
+
+    function commitAndClose(
+        address strikeSelection,
+        address optionsPremiumPricer,
+        uint256 premiumDiscount,
+        CloseParams calldata closeParams,
+        Vault.VaultParams calldata vaultParams,
+        Vault.VaultState calldata vaultState,
+        address collateralAsset
+    )
+        external
+        returns (
+            address otokenAddress,
+            uint256 premium,
+            uint256 strikePrice,
+            uint256 delta
+        )
+    {
+        uint256 expiry;
+
+        // uninitialized state
+        if (closeParams.currentOption == address(0)) {
+            expiry = getNextFriday(block.timestamp);
+        } else {
+            expiry = getNextFriday(
+                IOtoken(closeParams.currentOption).expiryTimestamp()
+            );
+        }
+
+        IStrikeSelection selection = IStrikeSelection(strikeSelection);
+
+        (strikePrice, delta) = closeParams.lastStrikeOverride ==
+            vaultState.round
+            ? (closeParams.overriddenStrikePrice, selection.delta())
+            : selection.getStrikePrice(expiry, false);
+
+        require(strikePrice != 0, "!strikePrice");
+
+        otokenAddress = getOrDeployOtoken(
```

Comment:
there are places like this that are different from vaultlifecycle - verifyOtoken is outside getOrDeployOToken. 

vaultlifecycle - [verifyOtoken is inside getOrDeployOToken ](https://github.com/ribbon-finance/ribbon-v2/blob/512b9573c8d71f717a460b6006236fc3ec29166f/contracts/libraries/VaultLifecycle.sol#L75)

this is a harmful difference. tests work no need to mess w it now